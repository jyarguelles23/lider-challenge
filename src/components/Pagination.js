import React, { Fragment, useState } from "react";

const Pagination = ({ setCurrentPage, totalPaginas }) => {
  const [paginaActual, setPaginaActual] = useState(0);

  const paginaAnterior = () => {
    const pagAnterior = paginaActual - 1;
    setPaginaActual(pagAnterior);
    setCurrentPage(paginaActual);
  };

  const paginaSiguiente = () => {
    const pagSiguiente = paginaActual + 1;
    setPaginaActual(pagSiguiente);
    setCurrentPage(paginaActual);
  };

  return (
    <Fragment>
      {paginaActual === 0 ? null : (
        <button
          onClick={paginaAnterior}
          type="button"
          className="btn btn-primary mr1"
          name="previous"
        >
          Anterior
        </button>
      )}

      {paginaActual === totalPaginas - 1 ? null : (
        <button
          onClick={paginaSiguiente}
          type="button"
          className="btn btn-primary mr1"
          name="next"
        >
          Siguiente
        </button>
      )}
    </Fragment>
  );
};

export default Pagination;
