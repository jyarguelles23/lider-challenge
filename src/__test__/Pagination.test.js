import Pagination from "../components/Pagination";
import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";

let tree = ReactTestRenderer;

const setCurrentPage = 0;
const totalPages = 3;

describe("<Pagination/>", () => {
  beforeEach(() => {
    tree = renderer.create(
      <Pagination setCurrentPage={setCurrentPage} totalPaginas={totalPages} />
    );
  });

  it("should render without crashing", () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render Next Button", () => {
    const next = tree.root;
    expect(next.findByProps({ name: "next" }).children).toEqual(["Siguiente"]);
  });

  it("should not render previous button", () => {
    const btn = tree.root;
    const button = btn.findByProps({ type: "button" }).children;
    expect(button).not.toBe(["Anterior"]);
  });
});
