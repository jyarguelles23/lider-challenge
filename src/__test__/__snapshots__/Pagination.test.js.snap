// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> should render without crashing 1`] = `
<button
  className="btn btn-primary mr1"
  name="next"
  onClick={[Function]}
  type="button"
>
  Siguiente
</button>
`;
